geocoding_android/android/src/main/java/com/baseflow/geocoding/MethodCallHandlerImpl.java

```diff
@@ -137,28 +149,40 @@ private void onPlacemarkFromAddress(final MethodCall call, final Result result)
 
         if (address == null || address.isEmpty()) {
             result.error(
-                "ARGUMENT_ERROR",
-                "Supply a valid value for the 'address' parameter.",
-                null);
+                    "ARGUMENT_ERROR",
+                    "Supply a valid value for the 'address' parameter.",
+                    null);
         }
 
         try {
-            final List<Address> addresses = geocoding.placemarkFromAddress(address);
+            geocoding.placemarkFromAddress(address, new GeocodeListenerAdapter() {
+
+                @Override
+                public void onGeocode(List<Address> addresses) {
+                    if (addresses != null && addresses.size() > 0) {
+                        result.success(AddressMapper.toAddressHashMapList(addresses));
+                    } else {
+                        result.error(
+                                "NOT_FOUND",
+                                String.format("No coordinates found for '%s'", address),
+                                null);
+                    }
+                }
+
+                @Override
+                public void onError(String errorMessage) {
+                    result.error(
+                            "NOT_FOUND",
```

Comment:
Practically these will always be IO errors, so I think it would be better to rename this to `IO_ERROR` to prevent confusion:
```suggestion
                            "IO_ERROR",
```